ui-admin/src/portal/siteContent/TranslationModal.tsx

```diff
@@ -1,22 +1,31 @@
 import React from 'react'
 import Modal from 'react-bootstrap/Modal'
 import { SiteContent } from '@juniper/ui-core'
-import { extractAllTexts, languageExtractToCSV } from './siteContentLanguageUtils'
+import { languageExtractToCSV, languageImportFromCSV } from './siteContentLanguageUtils'
 import { saveBlobAsDownload } from 'util/downloadUtils'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faDownload } from '@fortawesome/free-solid-svg-icons'
+import { faDownload, faUpload } from '@fortawesome/free-solid-svg-icons'
 import { Button } from '../../components/forms/Button'
+import { useFileUploadButton } from '../../util/uploadUtils'
 
 /** renders a modal for selecting translation functions */
-const TranslationModal = ({ onDismiss, siteContent }: {
-  siteContent: SiteContent, onDismiss: () => void
+const TranslationModal = ({ onDismiss, siteContent, setSiteContent }: {
+  siteContent: SiteContent, setSiteContent: (siteContent: SiteContent) => void, onDismiss: () => void
 }) => {
   const extractTexts = () => {
-    const extracts = extractAllTexts(siteContent)
-    const csvString = languageExtractToCSV(extracts)
+    const csvString = languageExtractToCSV(siteContent)
     const blob = new Blob([csvString], { type: 'text/plain' })
     saveBlobAsDownload(blob, 'site-translations.csv')
   }
+  const { FileChooser } = useFileUploadButton(file => {
+    const reader = new FileReader()
+    reader.onload = () => {
+      const updatedContent = languageImportFromCSV(siteContent, reader.result as string)
+      setSiteContent(updatedContent)
+    }
+    reader.readAsText(file)
```

Comment:
ux nit: I was surprised when I uploaded a file and nothing immediately visually happened. I thought an error occurred until I noticed my changes were applied after flipping to the other languages. Even though it's happy-path only, I think we should go ahead and add a loading indicator and a success toast, even if for developer-only convenience.